transfer/basic_download.go

```diff
@@ -203,7 +205,11 @@ func (a *basicDownloadAdapter) download(t *Transfer, cb TransferProgressCallback
 	}
 	written, err := tools.CopyWithCallback(dlFile, hasher, res.ContentLength, ccb)
 	if err != nil {
-		return fmt.Errorf("cannot write data to tempfile %q: %v", dlfilename, err)
+		wrappedErr := fmt.Errorf("cannot write data to tempfile %q: %v", dlfilename, err)
+		if errutil.IsRetriableError(err) {
+			return errutil.NewRetriableError(wrappedErr)
+		}
+		return wrappedErr
```

Comment:
Right now you'll have to check to make sure that the error isn't already retriable, like so:

```go
written, err := tools.CopyWithCallback(...)
if err != nil {
        err = errors.Wrapf(err, "cannot write data to tempfile %q", dlfilename)

        if errutil.IsRetriableError(err) {
                return err
        }

        return errors.NewRetriableError(err)
}

// ...
```